### PR TITLE
docs(general): clarify introduction section

### PR DESCRIPTION
**Pull request by niklas-wortmann**
I started to go over the general information section to remove some technical bla bla. It aims to make it easier for the reader to understand the important concepts and focus on the relevant parts!

Additionally, I slightly changed the sidebar and split the guides in an "Essential" and an "Advanced" section. We could add a small paragraph somewhere describing this structure to give an overview, but I think for now I can imagine this to help people to get into this by focusing on the essentials. At the same time, I split some larger sections into multiple smaller one and put them in the related category. E.g. the former `operators` guide is split into `operators`, `marble diagrams` & `higher-order Observables`

### CODE REVIEW

File: docs_app/content/guide/installation.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ To import only what you need using pipeable operators:
 import { of } from 'rxjs';
 import { map } from 'rxjs/operators';
 
-of(1,2,3).pipe(map(x => x + '!!!')); // etc
+of(1,2,3).pipe(map(x => x + '!!!'));
```
Comment (jakovljevic-mladen) [inline docs_app/content/guide/installation.md:25]: Hi there. There is an empty space left after `;` which might be removed. The other one couple of lines below is removed.

File: docs_app/content/guide/overview.md [hunk 2]
```diff
@@ -1,127 +1,50 @@
...
 
-fromEvent(document, 'click').subscribe(() => console.log('Clicked!'));
+console.log(c);
 ```
 
-### Purity
+Generally, considering usual programming paradigms the code above will log `20`. In a fully reactive approach the code would consider the change in the variable `a` and therefore log `30` instead. 
 
-What makes RxJS powerful is its ability to produce values using pure functions. That means your code is less prone to errors.
+This is only an example demonstrating the behavior of reactive programming.
```
Comment (jakovljevic-mladen) [inline docs_app/content/guide/overview.md:16]: This file has also couple of empty spaces at the end of some lines. If this is not important, please let me know, I won't report it anymore.

Comment (niklas-wortmann) [reply docs_app/content/guide/overview.md:16]: I don't have really strong opinions about that. IMO it helps maintaining the markdown file

File: docs_app/content/guide/installation.md (hunk 1 above)
Comment (niklas-wortmann) [reply docs_app/content/guide/installation.md:25]: thanks for spotting this

File: docs_app/content/guide/overview.md (hunk 2 above)
Comment (jakovljevic-mladen) [reply docs_app/content/guide/overview.md:16]: I kinda don't like empty spaces at the end of lines, they look a bit dirty to me. But if they can help, sure, keep having them.

File: docs_app/content/guide/observer.md [hunk 3]
```diff
@@ -29,13 +29,13 @@ const observer = {
 };
 ```
 
-When subscribing to an Observable, you may also just provide the callbacks as arguments, without being attached to an Observer object, for instance like this:
+When subscribing to an Observable, you can also provide the callbacks as anonymous function, without wrapping it in an object:
```
Comment (joshribakoff) [inline docs_app/content/guide/observer.md:32]: Whether or not you pass an anonymous function to the argument, does not matter, i don't think.  https://rxjs-ftnxyh.stackblitz.io

Could we consider rewording to "pass a references to callback functions as the arguments"?

Comment (niklas-wortmann) [reply docs_app/content/guide/observer.md:32]: good point. Thanks a lot for your input on that. Happy to use this wording

**Comment by benlesh** [comment]
@niklas-wortmann  we need to think of a way to merge this work with the work I'm doing over on #5592

**Comment by niklas-wortmann** [comment]
@benlesh as #5592 is going in master and will therefore be the updated docs for v7 and upcoming I would suggest moving this to the v6 branch to add some clarifications there.

**Comment by benlesh** [comment]
Core Team: this goes to v6 docs.

**Comment by benlesh** [comment]
Looks like this needs rebased.

**Comment by niklas-wortmann** [comment]
closed by #5871

**Pull request closed without merging**